app/models/discovery_rule.rb

```diff
@@ -36,13 +36,31 @@ class DiscoveryRule < ApplicationRecord
     end
   }
 
+  def unique_priority
+    Taxonomy.enabled_taxonomies.each do |enabled_taxonomy|
+      discovery_rule_ids  =  TaxableTaxonomy.where(
+        taxable_type: 'DiscoveryRule',
+        taxonomy_id: send("#{enabled_taxonomy.singularize}_ids")).pluck(:taxable_id)
+      discovery_rules = DiscoveryRule.unscoped.where(id: discovery_rule_ids)
+      if discovery_rules.pluck(:priority).include?(priority) && priority_changed?
+        errors.add(:priority, _("is already taken"))
+      end
```

Comment:
Can this be written without plucking and fetching all priorities? Like DiscoveryRule.unscoped.where(id: discovery_rule_ids, priority: priority)

Anyway, please move the `priority_changed?` to the very top and exit the method early if priority field is unchanged, this saves several SQL queries on every update.